File: contribution_tracker/src/core/domain/entities/user-commit.entity.ts
import { Column, DataType, Model, Table } from "sequelize-typescript";

@Table({
    tableName: "user_commits",
})
export default class UserCommitEntity extends Model {
    @Column({
        type: DataType.STRING(36),
        primaryKey: true,
        field: "id",
    })
    id!: string;

    @Column({
        type: DataType.BIGINT,
        field: "user_id",
    })
    userId!: number;

    @Column({
        type: DataType.STRING(500),
        field: "github_url",
    })
    githubUrl?: string;

    @Column({
        type: DataType.STRING(100),
        field: "github_sha",
    })
    githubSha?: string;

    @Column({
        type: DataType.STRING(255),
        field: "github_access_token",
    })
    githubAccessToken?: string;

    @Column({
        type: DataType.STRING(100),
        field: "github_login_name",
    })
    githubLoginName?: string;

    @Column({
        type: DataType.INTEGER,
        field: "user_consent",
    })
    userConsent!: number;

    @Column({
        type: DataType.STRING(100),
        field: "user_state",
    })
    userState!: string;

    @Column({
        type: DataType.INTEGER,
        field: "total_user_commits",
    })
    totalUserCommits?: number;

    @Column({
        type: DataType.DATE,
        field: "created_at",
    })
    createdAt?: Date;

    @Column({
        type: DataType.DATE,
        field: "updated_at",
    })
    updatedAt?: Date;
}

File: migrate/contribution_tracker_java_version/src/main/java/ct/contribution_tracker/core/domain/entity/UserCommitEntity.java
package ct.contribution_tracker.core.domain.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Table(name = "user_commit")
@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class UserCommitEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    private Integer userId;
    private String githubUrl;
    private String githubSha;
    private String githubAccessToken;
    private String githubLoginName;
    private Boolean userConsent;
    private String userState;
    private String platform;
}
